### PR TITLE
Window Titles Track Info

### LINKED ISSUE #387: Taskbar no longer shows Artist/track info on hover
OS: Windows 7 Enterprise (64-bit) SP1

Issue: Taskbar no longer shows Artist/track info on hover

Steps to Reproduce: Hover over the taskbar icon. The tooltip (including the previous/pause/next controls) displays, but rather than the artist/track info, just the "Google Play Music Desktop Player" text appears. I have the "Show song information always" setting checked. No hotkeys specified. Minimize to Tray is not set either.

gpmdp

### PR DESCRIPTION
**Pull request by jostrander**
Fixes #387 

Setting must be configured in the settings file, new property `enableWindowTitleInfo`.

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
> whoops, missed the main file...

This commit message is something I don't want to see in a PR, please squash your commits and force push

File: src/main/features/core/windowTitleInfo.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const windowTitle = WindowManager.get(global.mainWindowID).getTitle();
+let songInfo = {};
+
+if (Settings.get('enableWindowTitleInfo')) {
+  PlaybackAPI.on('change:song', (_songInfo) => {
+    songInfo = _songInfo;
```
Comment (MarshallOfSound) [inline src/main/features/core/windowTitleInfo.js]: Why are we doing this?

File: src/main/features/core/windowTitleInfo.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    WindowManager.get(global.mainWindowID).setTitle(
+      (_songInfo.title && _songInfo.artist) ? `${songInfo.title} - ${songInfo.artist}` : windowTitle
```
Comment (MarshallOfSound) [inline src/main/features/core/windowTitleInfo.js]: These two variables will **never** not be set

File: src/main/features/core/windowTitleInfo.js (hunk 1 above)
Comment (jostrander) [inline src/main/features/core/windowTitleInfo.js]: Saving the song info for state change events, such as play/pause so when the player is paused or stopped the player reverts to the regular title, and when playing again it changes to the Title - Artist. Could change to use PlaybackAPI calls directly.

File: src/main/features/core/windowTitleInfo.js (hunk 2 above)
Comment (jostrander) [inline src/main/features/core/windowTitleInfo.js]: Valid point, I needed to check if null, not if set. Will fix.

Comment (MarshallOfSound) [inline src/main/features/core/windowTitleInfo.js]: They will never be null either?

File: src/main/features/core/windowTitleInfo.js (hunk 1 above)
Comment (MarshallOfSound) [inline src/main/features/core/windowTitleInfo.js]: Yep, I would prefer using `PlaybackAPI.getCurrentSong()` (or whatever it actually is)

File: src/main/features/core/windowTitleInfo.js (hunk 2 above)
Comment (jostrander) [inline src/main/features/core/windowTitleInfo.js]: They won't be null, but they can be empty. Just figured that one out. I have a few tracks without artist names, but if its that far out as a use case I'll just remove the check.

Comment (MarshallOfSound) [inline src/main/features/core/windowTitleInfo.js]: Are those songs you uploaded your self?

I'd much rather go for
```js
(songInfo.title || 'Unknown Song')
(songInfo.artist || 'Unknown Artist')
```

kind of thing

**Comment by jostrander** [comment]
Squashed it down to one commit, tried getting rid of the rebase commit but it wasn't happening. If that doesn't work feel free to clone this and commit it yourself however it's got to happen and I'll just delete my branch.

**Comment by MarshallOfSound** [comment]
Have you tried running  
`git rebase -i origin/master`

You should then be able to mark all commits as "fixup"

**Comment by jostrander** [comment]
I was trying to use the HEAD syntax on a soft reset, but I just wasn't going back far enough apparently, committing now

**Comment by jostrander** [comment]
That didn't fix it either, even though my git log didn't show the rebase commit anymore. 

`git rebase -i origin/master` was of no help, probably would have been had I done it correctly in the first place.

**Comment by jostrander** [comment]
Alright, commits squashed, does this all look good now?

**Comment by MarshallOfSound** [comment]
Looks legit

**Pull request merged**